File: tugas_Heap/heap_py/data_entry.py
class DataEntry:
    """Merepresentasikan satu entri data dengan id dan nama."""

    def __init__(self, id_data: int, nama: str):
        self.id = id_data
        self.nama = nama

    def get_id(self) -> int:
        return self.id

    def get_nama(self) -> str:
        return self.nama

    def set_nama(self, nama: str) -> None:
        self.nama = nama

    def compare_to(self, other: "DataEntry") -> int:
        """Perbandingan untuk Min-Heap: id kecil lebih prioritas."""
        return self.id - other.id

    def reverse_compare_to(self, other: "DataEntry") -> int:
        """Perbandingan untuk Max-Heap: id besar lebih prioritas."""
        return other.id - self.id

    def to_long_string(self) -> str:
        return f"ID: {self.id}, Nama: {self.nama}"

    def __str__(self) -> str:
        return f"{self.id:<5} | {self.nama:<30}"

File: tugas_Heap/src/DataEntry.java
/**
 * DataEntry class - Merepresentasikan satu entri data dengan id dan nama
 * Digunakan untuk Min-Heap dan Max-Heap
 */
public class DataEntry implements Comparable<DataEntry> {
    private int id;
    private String nama;

    /**
     * Constructor
     */
    public DataEntry(int id, String nama) {
        this.id = id;
        this.nama = nama;
    }

    /**
     * Getter untuk id
     */
    public int getId() {
        return id;
    }

    /**
     * Getter untuk nama
     */
    public String getNama() {
        return nama;
    }

    /**
     * Setter untuk nama
     */
    public void setNama(String nama) {
        this.nama = nama;
    }

    /**
     * CompareTo - untuk Min-Heap (ascending by id)
     */
    @Override
    public int compareTo(DataEntry other) {
        return Integer.compare(this.id, other.id);
    }

    /**
     * Reverse compare - untuk Max-Heap (descending by id)
     */
    public int reverseCompareTo(DataEntry other) {
        return Integer.compare(other.id, this.id);
    }

    /**
     * ToString untuk display
     */
    @Override
    public String toString() {
        return String.format("%-5d | %-30s", id, nama);
    }

    /**
     * Untuk display dengan format lebih panjang
     */
    public String toLongString() {
        return "ID: " + id + ", Nama: " + nama;
    }
}
